components/Banner/examples/supplemental/MDCBannerViewLayout.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+typedef NS_ENUM(NSInteger, MDCBannerViewLayoutMode) {
+  MDCBannerViewLayoutSingleLineStyle = 0,              // All elements lays on the same line
+  MDCBannerViewLayoutMultiLineStackedButtonStyle = 1,  // Stacked button layout
+  MDCBannerViewLayoutMultiLineAlignedButtonStyle = 2   // All buttons lay on the same line
+};
```

Comment:
I'd expect these enums to be reactive rather than prescriptive. E.g. depending on the size and amount of content we'll need to use a different layout style.

As such I'd expect this to be a private implementation detail of the layout component and not part of the layout's API.

Reply:
It's a good point. Originally it was internal since it is dependent on the sizeToFit and was only used internally. The reason the `style` property was exposed was because (1). the view using the layout could utilize this property for some purposes. (not necessary for now) (2). it was mentioned that designer probably want the client to actively the stacked button or aligned button style in the future, so the exposed style property could be used for that. 

Will move it to the implementation file for now since we could easily expose it when needed.

Reply:
Addressed in 2b23608